### ISSUE TITLE
Cannot run seq2seq train example :(

### ISSUE THREAD

**Issue by LysSanzMoreta** [google/flax#74]
Hi!
First, thank you very much for this project!!!
Second, I am just simply trying to run this script to learn how to use GRU/LSTM: https://github.com/google/flax/blob/prerelease/examples/seq2seq/train.py

and I have encountered 2 issues:

1) **batch_metrics** is missing : Although is not an important issue, I just commented it out, but maybe you would like to do something about it :)

2) Error that is out of my skills:

```
Traceback (most recent call last):
  File "/.../FlaxExample.py", line 288, in <module>
    app.run(main)
  File "/.../anaconda3/lib/python3.7/site-packages/absl/app.py", line 300, in run
    _run_main(main, args)
  File "/.../anaconda3/lib/python3.7/site-packages/absl/app.py", line 251, in _run_main
    sys.exit(main(argv))
  File "/.../FlaxExample.py", line 284, in main
    _ = train_model()
  File "/.../FlaxExample.py", line 271, in train_model
    model = create_model()
  File "/.../FlaxExample.py", line 174, in create_model
    ((1, get_max_output_len(), vocab_size), jnp.float32)])
  File "/.../anaconda3/lib/python3.7/site-packages/flax/nn/base.py", line 183, in wrapper
    return super_fn(*args, **kwargs)
  File "/.../anaconda3/lib/python3.7/site-packages/flax/nn/base.py", line 392, in init_by_shape
    return jax_utils.partial_eval_by_shape(lazy_init, input_specs)
  File "/.../anaconda3/lib/python3.7/site-packages/flax/jax_utils.py", line 94, in partial_eval_by_shape
    output_shapes = jax.eval_shape(lazy_fn, *input_structs)
  File "/.../anaconda3/lib/python3.7/site-packages/jax/api.py", line 2023, in eval_shape
    out = pe.abstract_eval_fun(fun.call_wrapped, *map(abstractify, args_flat))
  File "/.../anaconda3/lib/python3.7/site-packages/jax/interpreters/partial_eval.py", line 256, in abstract_eval_fun
    instantiate=True)
  File "/.../anaconda3/lib/python3.7/site-packages/jax/interpreters/partial_eval.py", line 337, in trace_to_jaxpr
    jaxpr, (out_pvals, consts, env) = fun.call_wrapped(pvals)
  File "/.../anaconda3/lib/python3.7/site-packages/jax/linear_util.py", line 152, in call_wrapped
    ans = self.f(*args, **dict(self.params, **kwargs))
  File "/.../anaconda3/lib/python3.7/site-packages/jax/linear_util.py", line 152, in call_wrapped
    ans = self.f(*args, **dict(self.params, **kwargs))
  File "/.../anaconda3/lib/python3.7/site-packages/flax/jax_utils.py", line 88, in lazy_fn
    master = leaves[0]._trace.master  # pylint: disable=protected-access
  File "/.../anaconda3/lib/python3.7/site-packages/jax/core.py", line 365, in __getattr__
    attr = getattr(self.aval, name)
AttributeError: 'ShapedArray' object has no attribute '_trace'
```
 I believe I have the latest Jax and Flax versions:
Flax: 0.0.1a0
Jax : 0.1.58

Thanks for your help :)

**Comment by levskaya** [comment]
Sorry for the trouble as we clean things up for release - the former issue sounds like a simple bug, but can you see if the latter issue reproduces when using the newest jax version 0.1.59?  There were some internal changes between jax 0.1.58 and 0.1.59 that I think are causing this.

**Comment by levskaya** [comment]
your first issue is hopefully fixed by https://github.com/google/flax/pull/75

**Comment by LysSanzMoreta** [comment]
Hi! Thanks for the quick reply . I updated actually to 0.1.4 and now it's working. I see I will have a problem now because I am trying to use jax, flax and numpyro together. And numpyro requires a downgrade on jax version :( . I will see what I can do. Thanks again!!!!

**Comment by levskaya** [comment]
LysSanzMoreta - I might close this issue now after fixing the remaining small issue by merging #75 Please do get in touch again if you have any further issues on our end!

**Comment by LysSanzMoreta** [comment]
Well...Hi again! Thanks for everything! :) Actually, I re-updated to jax 1.41 and also reupdated flax but, when I run the newly merged seq2seq train.py I get a new error. The scan_in_dim is not where it should. I can see this issue #65 , but it does not seem like it was actually fixed... 
```
....
  File "/.../FlaxExample.py", line 121, in apply
    carry, _ = jax_utils.scan_in_dim(
AttributeError: module 'flax.jax_utils' has no attribute 'scan_in_dim'
```

Thanks again! :) May I ask when is it predicted to have the release or something a little bit more stable?

**Comment by avital** [comment]
Hi @LysSanzMoreta -- we are quickly ramping up our stability guarantees. 
Going forward, we will describe API changes in CHANGELOG.md.

We released [0.1.0rc2](https://pypi.org/project/flax/0.1.0rc2/#files) yesterday where `flax.jax_utils.scan_in_dim` is defined.

Depending on your appetite for being on the cutting edge, you could also consider pip installing flax directly from github, which would let you get updates the moment we push them.

**Comment by LysSanzMoreta** [comment]
Thanks for the update! Now it's working :)

I will think about how "cutting edge" do I want to be then hehehe
